src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/model/IndividualNeedsScores.kt

```diff
@@ -3,6 +3,13 @@ package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.restapi.model
 import com.fasterxml.jackson.annotation.JsonIgnore
 import com.fasterxml.jackson.annotation.JsonProperty
 import io.swagger.v3.oas.annotations.media.Schema
+import kotlin.reflect.full.memberProperties
+
+// Extension function to check if all properties of a data class are non-null
+fun <T : Any> T.areAllValuesPresent(): Boolean {
+  return this::class.memberProperties
+    .all { property -> property.call(this) != null }
+}
```

Comment:
Do we definitely want to use reflection?
PNI calculation is quite intensive and this could further degrade performance

Reply:
I think this is fine, and leads to a cleaner implementation as we can add and remove parameters for a data class and it'll still work. The performance hit is as a result of the repeated API calls to Oasys no?